src/sonic-daemon-base/sonic_daemon_base/daemon_base.py

```diff
@@ -90,9 +90,12 @@ def log_debug(self, msg, also_print_to_console=False):
 class DaemonBase(object):
     def __init__(self):
         # Register our signal handlers
-        signal.signal(signal.SIGHUP, self.signal_handler)
-        signal.signal(signal.SIGINT, self.signal_handler)
-        signal.signal(signal.SIGTERM, self.signal_handler)
+        if not signal.getsignal(signal.SIGHUP):
```

Comment:
Suggest to add a comment above this explaining that a derived class may have already registered signal handlers for these signals. If so, we don't want to override the derived class.

Reply:
fixed, comment added